fineract-provider/src/main/java/org/apache/fineract/organisation/office/service/OfficeReadPlatformServiceImpl.java

```diff
@@ -210,6 +210,22 @@ public OfficeData retrieveOffice(final Long officeId) {
         }
     }
 
+    @Override
+    public OfficeData retrieveOfficeWithExternalId(String externalId) {
```

Comment:
Not to my liking to be honest. This should be solved by using JPQL + a projection. I wouldn't breed more native queries unnecessarily although I understand it was a copy paste.